src/lib/hal/services/HalClient.ts

```diff
@@ -230,7 +230,13 @@ export class DefaultHalClient implements HalClient {
     return this.httpClient.request(fullRequest).then((response) => {
       if (response.status >= 200 && response.status < 300) {
         if (typeof response.data === 'string') {
-          response.data = JSON.parse(response.data);
+          if (typeof response.data === 'string') {
```

Comment:
I assume this is because the API could return with a 204?

If so can then be rewitten to catch the status code 204?

Reply:
Yes, will do here same